test/object-store/util/sync/redirect_server.hpp

```diff
@@ -0,0 +1,282 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2024 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+#pragma once
+
+#if defined(REALM_ENABLE_SYNC) && defined(REALM_ENABLE_AUTH_TESTS)
+
+#include <realm/sync/network/http.hpp>
+#include <realm/sync/network/network.hpp>
+#include <realm/sync/network/websocket.hpp>
+#include <realm/util/future.hpp>
+#include <realm/util/uri.hpp>
+#include <external/json/json.hpp>
+
+#include <catch2/catch_all.hpp>
+
+#include <thread>
+
+namespace realm::sync {
+
+class RedirectingHttpServer {
+public:
+    RedirectingHttpServer(std::string redirect_to_base_url, std::shared_ptr<util::Logger> logger)
+        : m_redirect_to_base_url(std::move(redirect_to_base_url))
+        , m_logger(std::make_shared<util::PrefixLogger>("HTTP Redirector ", std::move(logger)))
+        , m_acceptor(m_service)
+        , m_server_thread([this] {
+            m_service.run_until_stopped();
+        })
+    {
+        m_acceptor.open(m_endpoint.protocol());
+        m_acceptor.bind(m_endpoint);
+        m_endpoint = m_acceptor.local_endpoint();
+        m_acceptor.listen();
+        m_service.post([this](Status status) {
+            REALM_ASSERT(status.is_ok());
+            do_accept();
+        });
+    }
+
+    ~RedirectingHttpServer()
+    {
+        m_acceptor.cancel();
+        m_service.stop();
+        m_server_thread.join();
+    }
+
+    std::string base_url() const
+    {
+        return util::format("http://localhost:%1", m_endpoint.port());
```

Comment:
Is this port the same before and after the redirect? Just wondering if redirect mode would work when running baas locally. (Not that this is a requirement, just curious).